spi/spi_master_slave/spi_master/spi_master.c

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2021 Michael Stoops. All rights reserved.
+ * Portions copyright (c) 2021 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
```

Comment:
I'd like feedback on this. I'm flexible on this point, but this is what I thought was appropriate. The copyright lines themselves are per recommendation of https://spdx.org/licenses/BSD-3-Clause.html. I figure as I'm the one to took the initiative to create these examples, I deserve that credit. I used another example file as a starting point, and meant to keep it looking similar, so I think Raspberry Pi deserves credit too.

Reply:
Is there a reason why these files contain `SPDX-License-Identifier: BSD-3-Clause` but not the text of the license themselves? I'm thinking they should probably have both.

Reply:
> I'd like feedback on this. I'm flexible on this point, but this is what I thought was appropriate. The copyright lines themselves are per recommendation of https://spdx.org/licenses/BSD-3-Clause.html. I figure as I'm the one to took the initiative to create these examples, I deserve that credit. I used another example file as a starting point, and meant to keep it looking similar, so I think Raspberry Pi deserves credit too.

Yup we're totally fine with that. It's your copyright, of which part is ours, since you based it on another example as a starting point.

Reply:
Glad to hear it. I've copied in the license terms and adjusted the formatting. Resolving.